purchase_sale_inter_company/models/purchase_order.py

```diff
@@ -186,3 +186,38 @@ def button_cancel(self):
         sale_orders.action_cancel()
         self.write({"partner_ref": False})
         return super().button_cancel()
+
+
+class PurchaseOrderLine(models.Model):
+    _inherit = "purchase.order.line"
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        """Sync lines between an open purchase and an open sale order"""
+        lines = super().create(vals_list)
+        for order in lines.order_id.filtered(lambda x: x.state == "purchase"):
+            sale = self.env["sale.order"].search(
+                [("auto_purchase_order_id", "=", order.id)]
```

Comment:
Why not adding a o2m for avoiding this ugly syntax?

Reply:
It looks like o2m relations are limited by the record rules and I couldn't access it even with `sudo()`. An alternative approach has been a compute field with `compute_sudo`